static/app/components/u2f/u2finterface.tsx

```diff
@@ -41,17 +43,30 @@ class U2fInterface extends React.Component<Props, State> {
     hasBeenTapped: false,
     deviceFailure: null,
     responseElement: null,
+    isSafari: false,
+    failCount: 0,
   };
 
   async componentDidMount() {
     const supported = this.props.isWebauthnSigninFFEnabled
       ? !!window.PublicKeyCredential
       : await u2f.isSupported();
-
     // eslint-disable-next-line react/no-did-mount-set-state
     this.setState({isSupported: supported});
 
-    if (supported) {
+    const isSafari =
+      navigator.userAgent.includes('Safari') && !navigator.userAgent.includes('Chrome');
+
+    if (isSafari) {
+      // eslint-disable-next-line react/no-did-mount-set-state
+      this.setState({
+        deviceFailure: 'safari error',
```

Comment:
nit: could we update this deviceFailure to say something like 'safari: requires interaction'? also, do we still need to do the failCount thing if we're rendering differently for safari now?

Reply:
Will do on the deviceFailure, I'd like to have the failCount still so if it fails after the initial failure, it'll show the correct error screen, otherwise because isSafari is always on, it'll keep showing the "sign in with webauthn" without letting the user know what their error is.

Reply:
ahh good point. sounds good.